### PR TITLE
Lifeboats Simple Edition

### PR DESCRIPTION
**Pull request by ihatethisengine**
## About The Pull Request
Lifeboats without too much feature creep.

Lifeboats are a round-end system used during evacuation on hijack. 'Lifeboats' refer to the two large escape pods located on the aft upper deck of the Almayer.

Once evacuation is called, traditional escape pods become active and lifeboats begin a 10-minute timer countdown. Marines can hold the area around the lifeboats (the "lifeboat lounge") by constructing defenses. Once the timer is complete, the bulkhead doors shut, the docking hatches are blasted off and the lifeboats launch. The Almayer **WILL NOT** automatically self-destruct after their launching.

Queen can prevent lifeboat from launching by interacting with the console inside. 

Not many other changes, no reworking of old systems. I (or someone else) may add more features in future PRs, but my goal with this PR is to keep lifeboats as simple as possible because previous PRs have been stuck in MR hell due to feature creep and overhaul of too many systems. The only feature I could not refuse: if the player gets stuck between the lifeboat and the hangar after launch, he is thrown into space.

## Why It's Good For The Game

Lifeboats are designed in concept to replace Self-Destruct, allowing Marines to go out in a last stand while defending an area. This is meant to be a properly-defensible last stand that is still able to be broken through by Xenos with enough force.

Overall, lifeboats should improve the hijack game and allow Marines to defend a final objective.

## Changelog

:cl: BlackCrystalic, Ghostsheet, ihatethisengine, Firartix, nauticall
expansion: Lifeboats have been given functionality and can now be used without staff intervention.
/:cl:

### CODE REVIEW

**Comment by Geevies** [comment]
code looks good, i'll see about getting this testmerged ASAP

**Comment by morrowwolf** [comment]
Basic functionality looks good, let's get this test merged!

**Comment by ihatethisengine** [comment]
Added an additional 10 seconds for loading onto the lifeboats after confirming evacuation (in case Marines don't follow the timer), made camera consoles work

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Both sets of doors should close when it lifts off. Feel free to close the inner doors slightly later than the inner door for coolness.

Check to make sure they work post-round and/or if no one is on them.

**Comment by ihatethisengine** [comment]
Doors are fixed. 

Lifeboats do work if no one is on them.

Lifeboats don't work post-round for some reason. No idea why and not sure if it matters.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Should still launch post round as we discussed.

The lifeboat wall cabinets need sprites while vending. As it stands they disappear.

**Comment by ihatethisengine** [comment]
All fixed

**Comment by ihatethisengine** [comment]
All fixed @morrowwolf

**Review by Agameofscones — APPROVED** [review]
mapping is :+1:, sprites are OK, but personally I'd add a small animation (even just a color change on the white circle)

**Comment by ihatethisengine** [comment]
Added some animation as @Agameofscones requested

**Comment by AnCopper** [comment]
Is people getting yeeted into space when they stand next to the airlock after the boats leave intended? I saw a queen get sucked into space and deleted due to that.

**Comment by ihatethisengine** [comment]
@AnCopper yes if they stuck in the umbilical section when lifeboat is leaving it throws them into space. Feature. Unless they weren't in the umbilical section...

File: code/modules/shuttle/lifeboats.dm [hunk 1]
```diff
@@ -45,6 +105,15 @@
...
 	roundstart_template = /datum/map_template/shuttle/lifeboat_starboard
 
+/obj/docking_port/stationary/lifeboat_dock/Initialize(mapload)
+	. = ..()
+	GLOB.lifeboat_almayer_docks += src
+
+/obj/docking_port/stationary/lifeboat_dock/Destroy(force)
+	if (force)
+		GLOB.lifeboat_almayer_docks -= src
+	. = ..()
```
Comment (Geevies) [inline code/modules/shuttle/lifeboats.dm:115]:
```suggestion
	return ..()
```

**Review by Geevies — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
code looks good other than these nitpicks, great job

**Review by Geevies — APPROVED** [review]
code

**Comment by ihatethisengine** [comment]
MapDiffBot2 broke for no reason so I guess just ignore it

File: code/game/gamemodes/cm_self_destruct.dm [hunk 2]
```diff
@@ -137,7 +140,12 @@ var/global/datum/authority/branch/evacuation/EvacuationAuthority //This is initi
 				L -= i
 				sleep(50) //Sleeps 5 seconds each launch.
 			sleep(300) //Sleep 30 more seconds to make sure everyone had a chance to leave.
-			ai_announcement("ATTENTION: Evacuation complete. Outbound lifesigns detected: [P.passengers ? P.passengers  : "none"].", 'sound/AI/evacuation_complete.ogg')
+			var/lifesigns = 0
+			lifesigns += P.passengers
+			var/obj/docking_port/mobile/lifeboat/lifeboat = SSshuttle.getShuttle("lifeboat1")
```
Comment (morrowwolf) [inline code/game/gamemodes/cm_self_destruct.dm:145]: This needs to check both lifeboats.

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Pull request merged**